self-service-registartion/src/main/java/com/msm/ssr/utils/ApplicationServerUtils.java

```diff
@@ -0,0 +1,68 @@
+package com.msm.ssr.utils;
+
+import com.msm.ssr.exceptions.ApplicationServerUrlException;
+
+import javax.management.*;
+import java.lang.management.ManagementFactory;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * @author riste.jovanoski
+ * @since 6/13/2017
+ */
+public class ApplicationServerUtils {
+
+    /**
+     * Gets the server url on which the application is running.
+     * It returns a list with exactly two elements in which the first element is
+     * the url with an IP address, and the second is the url with a hostname.
+     * If no address can be constructed, an exception is thrown.
+     *
+     * @param contextPath
+     * @return Url with hostname and url with ip addres as list
+     * @throws ApplicationServerUrlException
+     */
+    public static List<String> getServerUrl(String contextPath) throws ApplicationServerUrlException {
+        try {
+            InetAddress inetAddress = InetAddress.getLocalHost();
+            String host = inetAddress.getHostAddress();
+            String hostname = inetAddress.getHostName();
+            InetAddress[] addresses = InetAddress.getAllByName(hostname);
+
+            MBeanServer mbs = ManagementFactory.getPlatformMBeanServer();
+            Set<ObjectName> objects = mbs.queryNames(new ObjectName("*:type=Connector,*"), Query.match(Query.attr("protocol"), Query.value("HTTP/1.1")));
+            for (ObjectName obj : objects) {
+                String scheme = mbs.getAttribute(obj, "scheme").toString();
```

Comment:
The scheme and port instance can be reused. We don't have to create new instance in each iteration